File: 205-IsomorphicStrings/isomorphicStrings.java
public class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] map = new int[512];
        
        for (int i = 0; i< s.length(); i++) {
            if (map[s.charAt(i)] != map[t.charAt(i)+256]) return false;
            map[s.charAt(i)] = map[t.charAt(i)+256] = i+1;// should use i+1 to avoid default value 0 in map.
        }
        return true;
        
    }
}

File: 205-IsomorphicStrings/isomorphicStrings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.size()!=t.size())    return false;
        vector<int> mps(128);
        vector<int> mpt(128);
        
        for(int i=0;i<s.size();i++){
            if(mps[s[i]]!=mpt[t[i]])    return false;
            mps[s[i]]=i+1;
            mpt[t[i]]=i+1;
        }
        return true;
    }
};